core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -997,7 +1066,9 @@ private Releasable acquireLock(Term uid) {
 
     private long loadCurrentVersionFromIndex(Term uid) throws IOException {
         try (final Searcher searcher = acquireSearcher("load_version")) {
-            return Versions.loadVersion(searcher.reader(), uid);
+            long version = Versions.loadVersion(searcher.reader(), uid);
+            assert incrementLookupMetrics(version);
```

Comment:
why do we only care about index based version lookups and not memory based lookups?